cmake/NeuronTestHelper.cmake

```diff
@@ -0,0 +1,361 @@
+# ~~~
+# Utility functions for creating and managing integration tests comparing
+# results obtained with NEURON and, optionally, CoreNEURON in different modes
+# of operation.
+#
+# 1. nrn_add_test_group(NAME name
+#                       SUBMODULE some/submodule
+#                       MODFILE_DIRECTORY mod_file_dir
```

Comment:
OK, I might find it easier to explore this with a relevant example. My current understanding is that we have to have special treatment for the modfiles because they are a build-time dependency of `nrnivmodl` that must be declared properly and copied to the right place, whereas the HOC files are only used at test runtime and are, therefore, more similar to the other test scripts.

For https://github.com/neuronsimulator/ringtest, which has a `.hoc` file at the top level, this seems to work fine, but maybe in a test with a more complicated directory structure we will need to have a way to set these environment variables.

Reply:
> OK, I might find it easier to explore this with a relevant example.

channel-benchmark that you are currently working testing would require HOC_LIBRARY_PATH.

> whereas the HOC files are only used at test runtime and are, therefore, more similar to the other test scripts.

Thats correct.

Reply:
OK. I suggest we follow this up when we merge the channel-benchmark tests too. Because tests are already executed using `cmake -E env` then it is already possible to do something like
```cmake
nrn_add_test(COMMAND HOC_LIBRARY_PATH=foo/bar special ...)
```
but if this becomes too widespread/messy then we could hide it behind a dedicated `HOCFILE_DIRECTORY` like you said.